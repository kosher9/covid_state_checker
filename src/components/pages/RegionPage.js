import React from 'react';
import { useParams } from 'react-router-dom';
import { getMapUrl } from '../mapUtils';
import Regions from '../Regions';

const RegionPage = () => {
  let { countryName } = useParams();
  const imageUrl = getMapUrl(countryName);
  countryName = countryName.replace(/_/g, ' ');

  return (
    <>
      <div
        className="container mx-auto h-40 opacity-10"
        style={{
          backgroundImage: `url(${imageUrl})`,
          backgroundSize: 'cover',
          backgroundRepeat: 'no-repeat',
          backgroundPosition: 'center',
        }}
      />
      <Regions countryName={countryName} />
    </>
  );
};

export default RegionPage;
