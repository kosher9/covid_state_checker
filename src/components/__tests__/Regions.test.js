import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/configStore';
import RegionPage from '../pages/Home';

describe('Region page test', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RegionPage />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
