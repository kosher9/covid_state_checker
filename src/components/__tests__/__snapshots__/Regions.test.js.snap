// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test render test 1`] = `
Array [
  <div
    className="container mx-auto h-40 opacity-10"
    style={
      Object {
        "backgroundImage": "url(https://mapsvg.com/static/maps/geo-calibrated/world.svg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      }
    }
  />,
  <div
    className="container mx-auto mt-4 absolute top-10"
  >
    <h3
      className="text-black text-center text-bold text-base rounded bg-slate-100"
    >
      Covid state
    </h3>
    <input
      className="rounded border-rose-600 m-auto block mt-5"
      onChange={[Function]}
      placeholder="Search country "
      type="text"
    />
  </div>,
  <h3>
    Stats by country
  </h3>,
  <div>
    <div
      className="grid grid-cols-2 striped gap-2"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>,
]
`;
